scripts/core/editor3/components/handlePastedText.js

```diff
@@ -101,3 +108,54 @@ const atomicBlock = (data, entity) => new ContentBlock({
     characterList: List([CharacterMetadata.create({entity})]),
     data: data
 });
+
+/**
+ * @ngdoc method
+ * @name allowEditSuggestion
+ * @param {string} operation - one of insert, backspace or delete
+ * @returns {Boolean} True if the current text don't contains a noneditable suggestion.
+ * @description Check if the current text don't contain a noneditable suggestion.
+ */
+export function allowEditSuggestion(action) {
+    const {suggestingMode, editorState} = this.props;
+    const content = editorState.getCurrentContent();
+    const selection = editorState.getSelection();
+    const types = ['DELETE_SUGGESTION', 'ADD_SUGGESTION'];
+
+    if (selection.getStartOffset() !== selection.getEndOffset()) {
+        for (let i = selection.getStartOffset(); i < selection.getEndOffset(); i++) {
+            const key = getEntityKey(content, getEntityKeyByOffset(content, selection, i), types);
+
+            if (!allowEditForKey(content, key, suggestingMode)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    const keyBefore = getEntityKey(content, getEntityKeyByOffset(content, selection, -1), types);
+    const keyAfter = getEntityKey(content, getEntityKeyByOffset(content, selection, 0), types);
+
+    return allowEditForKey(content, keyBefore, suggestingMode) && (action === 'backspace' || action === 'insert') ||
```

Comment:
If every line can be wrapped in parentheses, do so. It will be more explicit what it is you are checking.